### PR TITLE
Flywheel states -> Flywheel

### PR DESCRIPTION
**Pull request by Nitay4**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.flywheel;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public enum FlywheelState {
+
+	IDLE(Rotation2d.fromRotations(40)),
+	SHOOTING(Rotation2d.fromRotations(60));
+
+	Rotation2d velocity;
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java]: final

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	FlywheelState(Rotation2d velocity) {
+		this.velocity = velocity;
+	}
+
+	public Rotation2d getPower() {
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java]:
```suggestion
	public Rotation2d getVelocity() {
```

Comment (greenblitz4590) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java]: get velocity

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+	private final Flywheel flywheel;
+
+	public FlywheelStateHandler(Robot robot) {
+		this.flywheel = robot.getFlywheel();
+	}
+
+	public Command setState(FlywheelState state) {
+		return flywheel.getCommandsBuilder().setVelocity(state.getPower());
+	}
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java]: add if velocity 0 return stop command

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package frc.robot.subsystems.flywheel;
+
+
```
Comment (greenblitz4590) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java]: remvoe one line space

**Review by greenblitz4590 — CHANGES_REQUESTED** [review]
remove things that souldnt be here

File: \ [hunk 5]
```diff
@@ -0,0 +1,6 @@
+Merge branch 'core' into flywheel
+# Please enter a commit message to explain why this merge is necessary,
```
Comment (bossyshrimp68) [inline \]: What is this

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by drorta — APPROVED** [review]

**Review by drorta — APPROVED** [review]
nice! also do me a cr on swerve superstructure

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package frc.robot.subsystems.flywheel;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Robot;
+
+public class FlywheelStateHandler {
+
+	private final Flywheel flywheel;
+
+	public FlywheelStateHandler(Robot robot) {
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java]:
```suggestion
	public FlywheelStateHandler(Flywheel flywheel) {
```

**Review by Yotam7 — APPROVED** [review]

**Review by hillelv — COMMENTED** [review]
add flywheel to the superstructure

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package frc.robot.subsystems.flywheel;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public enum FlywheelState {
+
+	DEFAULT(Rotation2d.fromRotations(40)),
+	SHOOTER_OUTTAKE(Rotation2d.fromRotations(30)),
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java]: erm, what the sigma?

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/flywheel/FlywheelStateHandler.java:1]: why do you have this if you don't use it

File: src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java (hunk 7 above)
Comment (hillelv) [reply src/main/java/frc/robot/subsystems/flywheel/FlywheelState.java]: why does outtake make it go a bit slower

**Review by Yotam7 — APPROVED** [review]

**Review by hillelv — APPROVED** [review]

**Pull request merged**